### ISSUE TITLE
CMake ignoring order of build dependencies on latest Travis macOS image (Xcode 9.4)

### ISSUE THREAD

**Issue by EiffL** [LSSTDESC/CCL#434]
@alexander-mead @elisachisari @philbull @damonge @c-d-leonard A few CCL builds have started to fail on Travis CI in the last week, for apparently no reason. This is due to an update of the default build environment for Mac OS X (see https://blog.travis-ci.com/2018-07-19-xcode9-4-default-announce ) which occurred on Tuesday, 31 July.

As far as I can tell, cmake was updated from version `3.9.4` to version `3.11.3`, and it looks like it's not taking into account some `add_dependency` instructions that are supposed to tell it for instance to build GSL before generating the SWIG interface. The build order seems to be non-deterministic, which is why one of David's build worked and not the other in this example: https://travis-ci.org/LSSTDESC/CCL/builds/413097151

Obviously, this needs to be fixed and I'm looking into it, but in the meantime, here is what you can do to fix your travis builds: specifying the OSX image to use, reverting the previous default (Xcode 8.3). I have done it in the `halo_model` branch and it works: https://travis-ci.org/LSSTDESC/CCL/builds/413299475
Here is the commit:
https://github.com/LSSTDESC/CCL/commit/b81debc201c9be3ebe5f6ad464341f26d2cc10e9

**Comment by EiffL** [comment]
Ok, fixed. Turned out that CMake subtlety changed what was covered by a call to `add_dependency` on the swig interface, the `.so` file was indeed requiring the SWIG executable and GSL in order to be built, but the python interface file was not covered. I changed the way the dependencies are specified and now everything works on whatever cmake version. Opening a PR.
Just for posterity, the following cmake command was super helpful in order to debug this problem:
`set_property(GLOBAL PROPERTY GLOBAL_DEPENDS_DEBUG_MODE 1)`, which prints out the dependency tree between all the targets,
